addOns/callhome/src/main/java/org/zaproxy/addon/callhome/ExtensionCallHome.java

```diff
@@ -0,0 +1,235 @@
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionCallHome";
+
+    protected static final String PREFIX = "callhome";
+
+    private static final String ZAP_CFU_SERVICE = "https://cfu.zaproxy.org/ZAPcfu";
+
+    private static final String ISSUE_FILE = "/etc/issue";
+
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionCallHome.class);
+
+    private static final String BACK_BOX_ID = "BackBox";
+
+    public enum OS {
+        WINDOWS,
+        LINUX,
+        MAC_OS,
+        KALI,
+        BACK_BOX,
+        UNKNOWN;
+
+        @Override
+        public String toString() {
+            if (MAC_OS.equals(this)) {
+                return "MacOS";
+            } else if (BACK_BOX.equals(this)) {
+                return BACK_BOX_ID;
+            }
+            // First letter capitalised, rest lower case
+            return this.name().substring(0, 1) + this.name().substring(1).toLowerCase(Locale.ROOT);
+        }
+    }
+
+    private HttpSender httpSender = null;
+
+    private static OS os;
+    private static Boolean onBackBox = null;
+
+    public ExtensionCallHome() {
+        super(NAME);
+        setI18nPrefix(PREFIX);
+    }
+
+    @Override
+    public void postInit() {
+        this.setAutoUpdateSupplier(this);
+    }
+
+    @Override
+    public void unload() {
+        this.setAutoUpdateSupplier(null);
+    }
+
+    private void setAutoUpdateSupplier(Supplier<ZapXmlConfiguration> supplier) {
+        // XXX Change to not use reflection after 2.12.0
+        ExtensionAutoUpdate extAu =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionAutoUpdate.class);
+        if (extAu != null) {
+            try {
+                Method setSupplierMethod =
+                        extAu.getClass().getMethod("setCheckForUpdatesSupplier", Supplier.class);
+                LOGGER.debug("Setting CheckForUpdates supplier: {}", supplier);
+                setSupplierMethod.invoke(extAu, supplier);
+            } catch (Exception e) {
+                LOGGER.debug(
+                        "Failed to set CheckForUpdates supplier - expected to fail at 2.11.0", e);
+            }
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    private JSONObject getMandatoryRequestData() {
+        JSONObject json = new JSONObject();
+        json.put("zapVersion", Constant.PROGRAM_VERSION);
+        json.put("os", getOS().toString());
+        json.put(
+                "osVersion",
+                System.getProperty("os.name") + " : " + System.getProperty("os.version"));
+        json.put("javaVersion", System.getProperty("java.version"));
+        json.put("zapType", ZAP.getProcessType().name());
+        json.put("container", Constant.isInContainer() ? Constant.getContainerName() : "");
+        return json;
+    }
+
+    public ZapXmlConfiguration getCheckForUpdatesData()
```

Comment:
This could be private now?

Reply:
I would prefer to keep it public. The news equivalent will be public (to be added once this has been merged) and you never know when someone will come up with a good usecase for usinging this somewhere else...

Reply:
OK.